src/terminal/terminal/ansi_escape_codes.h

```diff
@@ -0,0 +1,279 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+#ifndef _GUAC_TERMINAL_ANSI_ESCAPE_CODES_H
+#define _GUAC_TERMINAL_ANSI_ESCAPE_CODES_H
+
+
+/**
+ * The maximum number of parameters the CSI can have is 16.
+ */
+#define GUAC_TERMINAL_MAX_ARGUMENTS 16
+
+/**
+ * When the parameter '2' is preceded by the escape character                   
+ * with '[' and followed by the the character 'J' the entire
+ * display is cleared. Without the '2' the line is deleted
+ * only from the cursor to the end of the display. 
+ */
+#define GUAC_TERMINAL_ERASE_DISPLAY 2
```

Comment:
Similar to the documentation for this macro, it may be worth considering embedding part of the context of this constant in the constant name. For example, if this value only has the defined meaning within a CSI sequence, then `GUAC_TERMINAL_CSI_ERASE_DISPLAY` might be a more sensible name. If that `J` you mention is yet another specific type of CSI sequence, then `GUAC_TERMINAL_CSI_WHATEVER_J_IS_ERASE_DISPLAY` would make yet more sense.